scan-payment/src/main/java/com/getbouncer/scan/payment/ml/yolo/yolo.kt

```diff
@@ -0,0 +1,68 @@
+package com.getbouncer.scan.payment.ml.yolo
+
+import android.graphics.RectF
+import android.util.Size
+import com.getbouncer.scan.framework.util.indexOfMax
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import kotlin.math.exp
+
+fun sigmoid(x: Float): Float = (1.0f / (1.0f + exp((-x))))
+
+fun softmax(a: FloatArray): FloatArray {
+    var sum = 0f
+    for (f in a) {
+        sum += exp(f)
+    }
+    return a.map { exp(it) / sum }.toFloatArray()
+}
+
+/**
+ * Takes a layer from the raw YOLO model output and performs post-processing on it,
+ * returning a List<DetectionBox>
+ */
+fun processYoloLayer(
+    layer: Array<Array<FloatArray>>,
+    anchors: Array<Pair<Int, Int>>,
+    imageSize: Size,
+    numClasses: Int,
+    confidenceThreshold: Float
+): List<DetectionBox> {
+    val results = mutableListOf<DetectionBox>()
+    for (i in layer.indices) for (j in layer[i].indices) for (k in 0..2) {
+        val offset = (numClasses + 5) * k
+        var confidence = sigmoid(layer[i][j][offset + 4])
+        var confidenceClasses = FloatArray(numClasses)
+
+        for (m in 0 until numClasses) {
+            confidenceClasses[m] = layer[i][j][offset + 5 + m]
+        }
+        confidenceClasses = softmax(confidenceClasses)
```

Comment:
val confidenceClasses = (0 until numClasses).map { layer[i][j][offset + 5 + it] }.softMax2D()